server/src/main/java/org/opensearch/rest/action/list/RestIndicesListAction.java

```diff
@@ -0,0 +1,175 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.list;
+
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.support.GroupedActionListener;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.Table;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.core.common.Strings;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.action.RestResponseListener;
+import org.opensearch.rest.action.cat.RestIndicesAction;
+import org.opensearch.rest.action.cat.RestTable;
+import org.opensearch.rest.pagination.IndexBasedPaginationStrategy;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest.DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT;
+import static org.opensearch.rest.RestRequest.Method.GET;
+
+/**
+ * _list API action to output indices in pages.
+ *
+ * @opensearch.api
+ */
+public class RestIndicesListAction extends AbstractListAction {
+    private static final String DEFAULT_LIST_INDICES_PAGE_SIZE_STRING = "1000";
+    private static final String PAGINATED_ELEMENT_KEY = "indices";
+
+    @Override
+    public List<Route> routes() {
+        return unmodifiableList(asList(new Route(GET, "/_list/indices"), new Route(GET, "/_list/indices/{index}")));
+    }
+
+    @Override
+    public String getName() {
+        return "list_indices_action";
+    }
+
+    @Override
+    public boolean allowSystemIndexAccessByDefault() {
+        return true;
+    }
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_list/indices\n");
+        sb.append("/_list/indices/{index}\n");
+    }
+
+    @Override
+    public RestChannelConsumer doListRequest(final RestRequest request, final NodeClient client) {
+        final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
+        final boolean local = request.paramAsBoolean("local", false);
+        TimeValue clusterManagerTimeout = request.paramAsTime("cluster_manager_timeout", DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT);
+        final boolean includeUnloadedSegments = request.paramAsBoolean("include_unloaded_segments", false);
+        final String requestedToken = request.param("next_token");
+        final int pageSize = Integer.parseInt(request.param("size", DEFAULT_LIST_INDICES_PAGE_SIZE_STRING));
+        final String requestedSortOrder = request.param("sort", "ascending");
```

Comment:
This is the part that I think should be thinned out. Instead of receiving a `RestRequest` it should be a `PaginatedRestRequest` where a bunch of parameters were parsed outside of `RestIndicesListAction`.

Reply:
Thanks @dblock for the feedback. Makes sense to declutter this and also avoid duplication of code. Let me think through a bit more on, if and how can this be achieved.

Reply:
Have moved things to a common place now. `RestIndicesList` action is now directly extending `RestIndicesAction` and all the existing parameters are hence only getting parsed at one place. While, the pagination related params can be parsed individually by the list/paginated APIs.